### ISSUE TITLE
detox scroll action triggers a click if there is no place to scroll

### ISSUE THREAD

**Issue by Niryo** [wix/Detox#2510]
### Description
if we trigger a scroll on an element, but there is no place to scroll- instead of doing nothing, detox accidentally triggers a click. this lead to some un-expected behavior (in platform master we use deotx 17.8.3 and don't have this problem)

- [x] I have tested this issue on the latest Detox release and it still reproduces

#### Reproduction

try to trigger a scroll on a list of buttons (for example action sheet), when there is no actually room for scrolling. it will trigger a click on one of the buttons...

#### Expected behavior
scroll should not trigger a click. 

#### Environment (please complete the following information):
 - Detox: ^17.13.2
 - React Native: 0.62.2
 - Node:
 - Device: iphone 11

**Comment by LeoNatan** [comment]
This is a bug in RN’s gesture handler, not in Detox. The same issue doesn’t reproduce on native table view with selectable cells, for example. As a workaround, enable bouncing in the scroll view primary axis.

**Comment by LeoNatan** [comment]
@rotemmiz Reproduced the same issue with his finger, BTW, without any Detox. Open an issue with FB or fix in our internal fork.

**Comment by Niryo** [comment]
I can't reproduce on android, I will check with rotem, tnx

**Comment by Niryo** [comment]
update: checked on both android device and ios simulator and cannot reproduce on a real app. also, as I said, the same test is passing with detox 17.8.3 but failing with greater 17.13.2

**Comment by LeoNatan** [comment]
Please do not reopen this issue. There is nothing we can do. There might be some differences with some magic numbers that work in one Detox version and not the other (Detox 17 has more precise scrolling), but the fundamental issue is with the RN gesture handler.

**Comment by LeoNatan** [comment]
@Niryo If you are interested in a technical explanation of what the problem is, here it is:

iOS has many different specialized gesture recognizers, such as tap, pan, pinch, long press, etc.
With RN, FB reinvents the wheel 🙄 and uses one gesture recognizer for all gesture types. Gesture recognizers work with touches. The system provides the information such as touch began, touch moved, touch ended, and it's up to the gesture recognizer to figure out what is going on. On iOS, a regular tap gesture recognizer gets primed and a touch begins, but if touch moves (the user moved the finger beyond a well known threshold—10pt), the tap gesture is cancelled. This is why Detox never triggers taps on native UI. RN doesn't have this check. If you start a touch, move it and then end by lifting the finger, this counts as a "tap". You can reproduce this very easily by creating a large touchable opacity with a handler, put your finger on the top, move the finger to the bottom and lift the finger. The touchable opacity handler will fire, erroneously.

**Comment by Niryo** [comment]
Thanks for the detailed explanation. 
I think the issue could be different in my case, because the scroll starts and ends on different views, and I can't reproduce the trigger of the tap with my finger. I showed a video of the issue to @rotemmiz and he agreed that it looks like something that could be solvable by detox (could be hacky, but solvable). Anyways, I already fixed my failing test, I am just worried that other people's build will also fail after upgrading detox...
